### PR TITLE
[py] Honor cmd line args passed to Service ctor

### PR DESCRIPTION
**Pull request by jugglinmike**
Ensure that when command line arguments are specified during creation of
a Service object for GeckoDriver, these arguments are used in the
resulting invocation of the binary.

- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

### CODE REVIEW

File: py/test/selenium/webdriver/firefox/ff_service_tests.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    found = False
+
+    args = service.command_line_args()
+
+    for idx in xrange(len(args) - 1):
+        if args[idx] == "--log" and args[idx + 1] == "trace":
+            found = True
+            break
+
+    assert found, "Provided arguments exist in array"
```
Comment (Dude-X) [inline py/test/selenium/webdriver/firefox/ff_service_tests.py]: Error message should state `Provided arguments do not exist in array`.

**Comment by lmtierney** [comment]
@jugglinmike Looks good once you update the assertion message per @Dude-X

File: py/test/selenium/webdriver/firefox/ff_service_tests.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+from selenium.webdriver.firefox.service import Service
+
+def test_command_line_args():
+    service = Service("geckodriver", service_args=["--log", "trace"])
+    found = False
+
+    args = service.command_line_args()
+
+    for idx in xrange(len(args) - 1):
```
Comment (Dude-X) [inline py/test/selenium/webdriver/firefox/ff_service_tests.py]: `xrange` is not Python 3 compatible. Use `range`. It's not that bad for small lists in Python 2.

**Comment by jugglinmike** [comment]
Sure, I've updated the test code as requested by @Dude-X.

**Comment by lmtierney** [comment]
@jugglinmike can you move the test to `test/selenium/webdriver/marionette` as this only pertains to geckodriver? And rename the test file appropriately. Sorry, forgot about it and only noticed it while trying to test.

**Comment by jugglinmike** [comment]
No worries, @lmtierney. Test file renamed.

**Comment by lmtierney** [comment]
Merged, thanks! You know how to get ahold of @AutomatedTester if you need a point release on this

**Comment by jugglinmike** [comment]
Happy to help :)

**Comment by djowett** [comment]
@jugglinmike This looks good, but don't you also need to use get firefox.Webdriver() to pass service_args='something' to Service() ?

**Comment by jugglinmike** [comment]
@ djowett you can specify the option by importing the `Service` constructor through the `selenium.webdriver.firefox.service` package.

**Comment by djowett** [comment]
@jugglinmike thanks for the reply, but to do that you have to subclass and override `selenium.webdriver.firefox.webdriver.service` right? Or am I missing something?

**Pull request merged**